test/e2e/framework/framework.go

```diff
@@ -193,35 +195,18 @@ func (f *Framework) setupNamespace(ctx context.Context) {
 	}, metav1.CreateOptions{})
 	o.Expect(err).NotTo(o.HaveOccurred())
 
-	// Wait for user ServiceAccount.
-	By(fmt.Sprintf("Waiting for ServiceAccount %q in namespace %q.", userSA.Name, userSA.Namespace))
-	ctxUserSa, ctxUserSaCancel := watchtools.ContextWithOptionalTimeout(ctx, serviceAccountWaitTimeout)
-	defer ctxUserSaCancel()
-	userSA, err = WaitForServiceAccount(ctxUserSa, f.KubeAdminClient().CoreV1(), userSA.Namespace, userSA.Name)
+	// Send a TokenRequest for user service account.
+	userTokenRequest, err := f.KubeAdminClient().CoreV1().ServiceAccounts(ns.Name).CreateToken(ctx, userSA.Name, &authenticationv1.TokenRequest{
+		Spec: authenticationv1.TokenRequestSpec{
+			ExpirationSeconds: pointer.Int64(int64(24 * time.Hour / time.Second)),
```

Comment:
What do you mean by that it's dangerous? Duration is an int64 nanosecond count, so there's nothing wrong with dividing it. It's not a valid Duration after that of course, but that's why the int64 cast. `func (d *Duration) Seconds() float64` does the same thing partially, except it produces a float64 and then you truncate it anyway.

Reply:
> It's not a valid Duration after that of course

That's the reason why it's dangerous, you get time.Duration which needs special handling.
There're tools in a library to do it right and devs should use them. These functions were introduced later than initial implementation, probably because people were doing it your way. I recall I debugged simillar code so now I'm extra careful about time.Duration division.

Reply:
Well, I agree that it's more bug proof to use the lib function. I've sent a fix.